File: trunk/doc/Examples_SPL/Expresiones/ProductB-ShortCircuitPostFix/Add.cs
using System;
using Expresiones;

namespace ProductB_ShortCircuitPostFix
{
    public class Add : BinaryOperator, IAddShortEval, IAddPosfix
    {
        public Add(IExpressionShortCircuitPostFix op1, IExpressionShortCircuitPostFix op2) {
            this.mixinBinaryOperatorPosfix = new AddPosfix(op1,op2);
            this.mixinBinaryOperatorShortEval = new AddShortEval(op1, op2);
        } // add
        
    }
}

File: doc/Examples_SPL/expressionsCaesarJ/src/eval/Add.java
package eval;

cclass eval.Eval;

public cclass Add extends BinaryOperator {
	protected ExpressionEval op1;
	protected ExpressionEval op2;
	public AddEval(ExpressionEval op1, ExpressionEval op2){
		this.op1=op1;
		this.op2=op2;
	}
	public int eval(){
		return this.op1.eval()+this.op2.eval();
	}
}
